components/event-publisher/org.wso2.carbon.event.output.adapter.core/src/main/java/org/wso2/carbon/event/output/adapter/core/EventAdapterUtil.java

```diff
@@ -81,4 +94,42 @@ public static Map<String, String> getGlobalProperties(String type) {
         return null;
     }
 
+    /**
+     * Retrieves the access token using the client credentials grant type.
+     *
+     * @param clientId      The client ID.
+     * @param secret        The client Secret.
+     * @param tokenEndpoint The token endpointURL.
+     * @param scopes        The scopes to be requested.
+     * @return Access Token.
+     */
+    public static String getAccessToken(String clientId, String secret, String tokenEndpoint, String scopes) {
+
+        try (CloseableHttpClient httpClient = HttpClients.createDefault();
+             CloseableHttpResponse response = httpClient.execute(createTokenRequest(clientId, secret,
+                     tokenEndpoint, scopes))) {
+
+            String responseString = EntityUtils.toString(response.getEntity());
+            JSONObject jsonResponse = new JSONObject(responseString);
+            return jsonResponse.getString("access_token");
+        } catch (Exception e) {
+            LOG.error("Error while getting access token", e);
+            return null;
+        }
+    }
+
+    private static HttpPost createTokenRequest(String clientId, String secret, String tokenEndpoint, String scopes) {
+
+        HttpPost request = new HttpPost(tokenEndpoint);
+        request.setHeader("Content-Type", "application/x-www-form-urlencoded");
+
+        // Request body.
+        String body = "client_id=" + clientId +
```

Comment:
Instead of creating request body, check how we set the parameters

Reply:
List<NameValuePair> urlParameters = new ArrayList<>();
urlParameters.add(new BasicNameValuePair("token", authorizedAccessTokenContext.getRefreshToken()));